src/transaction.rs

```diff
@@ -57,26 +59,25 @@ impl FundingTransaction {
 pub struct CommitTransaction(Transaction);
 
 impl CommitTransaction {
-    // TODO: Handle expiry by passing it as an argument
     pub fn new(
         TX_f: &FundingTransaction,
-        (_X_0, _R_0, _Y_0): (PublicKey, RevocationPublicKey, PublishingPublicKey),
-        (_X_1, _R_1, _Y_1): (PublicKey, RevocationPublicKey, PublishingPublicKey),
-    ) -> Self {
-        Self(Transaction {
+        (X_0, R_0, Y_0): (PublicKey, RevocationPublicKey, PublishingPublicKey),
+        (X_1, R_1, Y_1): (PublicKey, RevocationPublicKey, PublishingPublicKey),
+        time_lock: u32,
+    ) -> Result<Self> {
+        let descriptor = Self::descriptor((X_0, R_0, Y_0), (X_1, R_1, Y_1), time_lock)?;
+
+        Ok(Self(Transaction {
             version: 2,
             lock_time: 0,
             input: vec![TX_f.as_txin()],
             // This output is the same as the input, except for the
             // spending conditions
             output: vec![TxOut {
                 value: TX_f.value().as_sat(),
-                script_pubkey: todo!(
-                    "Use Miniscript to generate the transaction output with three
-                 spending conditions using (X_0, R_0, Y_0) and (X_1, R_1, Y_1)"
-                ),
+                script_pubkey: descriptor.script_pubkey(),
```

Comment:
I'm confused if we need to use `.witness_script()` or `.script_pubkey()` (:

Reply:
Yeah I think `witness_script` goes with the witness stack when signing. To validate the transaction a node would then:
- hash the witness script to confirm it matches the script_pubkey
- run the witness script and stack to confirm the script outputs 1

I am like 80% sure :)